src/core/PWScore.cpp

```diff
@@ -2262,6 +2267,18 @@ bool PWScore::Validate(const size_t iMAXCHARS, CReport *pRpt, st_ValidateResults
       }
     }
 
+    // Test if passkey is complete
+    bool hasAllPasskeyFields = ci.IsPasskeyCredentialIDSet() && ci.IsPasskeyRelyingPartyIDSet()
```

Comment:
Could we define separate methods to calculate hasAllPasskeyFields and hasNonePasskeyFields? In general, `Validate` is quite long already, and it would be good to extract some of the logic.

Reply:
I've moved the logic to CItemData, see updated commit.

Better like this?